src/random_read_worker.rs

```diff
@@ -31,26 +29,29 @@ impl FileSourceQueryGenerator {
     pub fn new<P: AsRef<Path>>(query_file: P) -> io::Result<Self> {
         let query_file = query_file.as_ref();
 
-        let queries = std::fs::read_to_string(query_file)?
+        let queries: Vec<_> = std::fs::read_to_string(query_file)?
             .lines()
             .filter(|l| !l.is_empty())
             .map(ToOwned::to_owned)
             .collect();
 
-        Ok(Self { queries, ix: 0 })
+        Ok(Self { queries_original_order: queries.clone(), queries, ix: 0 })
     }
 }
 
 impl QueryGenerator for FileSourceQueryGenerator {
-    fn next_query(&mut self) -> Cow<str> {
+    fn next_query(&mut self) -> (Option<usize>, Cow<str>) {
         let cur_ix = self.ix;
         if cur_ix == 0 {
             self.queries.shuffle(&mut rand::thread_rng());
         }
 
         self.ix = (self.ix + 1) % self.queries.len();
 
-        Cow::Borrowed(&self.queries[cur_ix])
+        let ret_query = &self.queries[cur_ix];
+        let orig_ix = self.queries_original_order.iter().position(|q| q == ret_query).unwrap();
+
+        (Some(orig_ix), Cow::Borrowed(ret_query))
```

Comment:
do all workers now follow the same order?

Reply:
Not something to address in this PR, but having a fixed order might also come in handy at some point.

Reply:
Created an issue